x/concentrated-liquidity/lp.go

```diff
@@ -321,21 +321,19 @@ func (k Keeper) addToPosition(ctx sdk.Context, owner sdk.AccAddress, positionId
 		return 0, sdk.Int{}, sdk.Int{}, types.PositionSuperfluidStakedError{PositionId: position.PositionId}
 	}
 
-	// Withdraw full position.
-	amount0Withdrawn, amount1Withdrawn, err := k.WithdrawPosition(ctx, owner, positionId, position.Liquidity)
+	positions, err := k.GetAllPositionIdsForPoolId(ctx, types.PositionPrefix, position.PoolId)
```

Comment:
Doing this check and checking if this would be the last position before the actual `WithdrawPosition` rather than doing this after calling `WithdrawPosition` allows us to save computation

Reply:
Wait isn't this linear in the number of position ID's for a pool? Thats way too expensive to run on addToPosition?